### PR TITLE
vjudge: add hduoj

### PR DESCRIPTION
**Pull request by Lyricus233**
demo: https://bpoj.top/d/hduoj/p

### CODE REVIEW

**Comment by undefined-moe** [comment]
Please fix all CI issues before we can merge this pull request.

File: packages/vjudge/src/providers/hduoj.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                username: this.account.handle,
+                userpass: this.account.password,
+                login: 'Sign In',
+            });
+        return this.loggedIn;
+    }
+
+    async getProblem(id: string) {
+        logger.info(id);
+        const res = await superagent.get(`http://acm.hdu.edu.cn/showproblem.php?pid=${id.split('P')[1]}`)
```
Comment (pandadtdyy) [inline packages/vjudge/src/providers/hduoj.ts]: remove endPoint
```suggestion
        const res = await superagent.get(`/showproblem.php?pid=${id.split('P')[1]}`)
```

**Review by pandadtdyy — CHANGES_REQUESTED** [review]

**Comment by Lyricus233** [comment]
hduoj language settings:

```
hduoj:
  display: HDUOJ
  execute: /bin/echo Invalid
  domain:
  - hduoj
hduoj.0:
  display: G++
  monaco: cpp
  highlight: cpp astyle-c
  comment: //
hduoj.1:
  display: GCC
  monaco: c
  highlight: c astyle-c
  comment: //
hduoj.2:
  display: C++
  monaco: cpp
  highlight: cpp astyle-c
  comment: //
hduoj.3:
  display: C
  monaco: c
  highlight: c astyle-c
  comment: //
hduoj.4:
  display: Pascal
  monaco: pascal
  highlight: pascal
  comment: //
hduoj.5:
  display: Java
  monaco: java
  highlight: java astyle-java
  comment: //
hduoj.6:
  display: C#
  monaco: plain
  highlight: plain
```

File: packages/vjudge/package.json [hunk 2]
```diff
@@ -14,6 +14,8 @@
     },
     "devDependencies": {
         "@types/jsdom": "21.1.1",
+        "@types/superagent": "^4.1.16",
+        "superagent-charset": "^1.2.0",
```
Comment (undefined-moe) [inline packages/vjudge/package.json]: why devdeps?

**Review by undefined-moe — CHANGES_REQUESTED** [review]

File: packages/vjudge/package.json [hunk 3]
```diff
@@ -10,6 +10,8 @@
     "dependencies": {
         "@hydrooj/utils": "workspace:*",
         "jsdom": "^22.1.0",
+        "@types/superagent": "^4.1.16",
```
Comment (undefined-moe) [inline packages/vjudge/package.json]: why deps?

**Review by undefined-moe — CHANGES_REQUESTED** [review]

**Review by undefined-moe — APPROVED** [review]

**Pull request merged**